src/test/ui/trait-bounds/unsized-bound.stderr

```diff
@@ -0,0 +1,234 @@
+error[E0277]: the size for values of type `B` cannot be known at compilation time
+  --> $DIR/unsized-bound.rs:2:12
+   |
+LL | trait Trait<A> {}
+   |             - required by this bound in `Trait`
+LL | impl<A, B> Trait<(A, B)> for (A, B) where A: ?Sized, B: ?Sized, {}
+   |         -  ^^^^^^^^^^^^^ doesn't have a size known at compile-time
```

Comment:
This span is confusing to me. The trait in an impl block is not something that can have a size. I think we would want to point at `(A, B)` inside the `Trait<(A, B)>`.

Reply:
For what is worth that's the same span we currently use:

```
error[E0277]: the size for values of type `B` cannot be known at compilation time
 --> src/test/ui/trait-bounds/unsized-bound.rs:2:12
  |
1 | trait Trait<A> {}
  |             - required by this bound in `Trait`
2 | impl<A, B> Trait<(A, B)> for (A, B) where A: ?Sized, B: ?Sized, {}
  |         -  ^^^^^^^^^^^^^ doesn't have a size known at compile-time
  |         |
  |         this type parameter needs to be `std::marker::Sized`
  |
  = note: required because it appears within the type `(A, B)`
help: consider relaxing the implicit `Sized` restriction
  |
1 | trait Trait<A: ?Sized> {}
  |              ^^^^^^^^
```

And @Aaron1011 has been working piecemeal on HIR based checks that could give us the span you want, IIRC.

Reply:
gotcha!